modules/basic-skills/src/hooks/after_incoming_middleware/05_extract_slots.js

```diff
@@ -0,0 +1,48 @@
+const _ = require('lodash')
+
+// Make sure extractedSlots exists
+event.state.session.extractedSlots = event.state.session.extractedSlots || {}
+rotateTurns()
+extractIntentSlots()
+
+function extractIntentSlots() {
+  const slots = _.flatten(_.values(event.nlu.slots)).filter(x => !!x.value) // only non-null slots
+  for (let slot of slots) {
+    // BETA(11.8.4): Prevent overwrite of the slot if explicitely demanded
+    if (
+      event.state.session.extractedSlots[slot.name] &&
+      event.state.session.extractedSlots[slot.name].overwritable == false
+    ) {
+      continue
+    }
+
+    // Slot is an array when the NLU is confused about the results
+    // The array is sorted by confidence so we take the first index
+    if (Array.isArray(slot)) {
```

Comment:
Are you sure about this ? It would be nice to ask the user to fix the confusion

Reply:
Resolved discussion offline:  we will implement disambiguation at the core level